core/src/G3Timesample.cxx

```diff
@@ -5,43 +5,57 @@
 
 #include <container_pybindings.h>
 #include <G3Timesample.h>
-//#include <exceptions.h>
 
-// Templates for vector operations.  g3vectype might be, for example,
-// G3VectorDouble.  One could go all the way and SFINAE this... but
-// perhaps not everything should require a black belt.
 
+// Templates for vector operations.  These work on any std::vector,
+// and thus on any G3Vector.
+
+// Returns the vector size if the type matches, and -1 otherwise.
 template <typename g3vectype>
 inline
-int g3_vect_size(const G3FrameObjectPtr vp)
+ssize_t vect_size(const G3FrameObjectPtr &vp)
 {
 	auto v = boost::dynamic_pointer_cast<const g3vectype>(vp);
 	return v == nullptr ? -1 : v->size();
 }
 
-template <typename g3vectype>
+// Concatenates two compatible vectors.
+template <typename vectype>
 inline
-void g3_concat(g3vectype &output, const g3vectype &src1, const g3vectype &src2)
+vectype vect_concat(const vectype &src1, const vectype &src2)
```

Comment:
Again might want to be careful here with unnecessary copy on return.

Reply:
Well, this is an inline function.  But I changed it in latest commit, anyway.

The C++11 solution for this, generally, is "move constructors".  I have no idea whether those are implemented for G3Vector, perhaps through inheritance from std::vector.